File: backend/src/main/java/com/example/backend/Program/ProgramRepository.java
package com.example.backend.Program;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Repository;

@Repository
public class ProgramRepository {
    private final JPAProgramRepository programRepository;

    @Autowired
    public ProgramRepository(JPAProgramRepository programRepository) {
        this.programRepository = programRepository;
    }
}

File: backend/src/main/java/com/example/backend/exercise_generator/repository/ProgramRepository.kt
package com.example.backend.exercise_generator.repository

import com.example.backend.user.model.Program
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface ProgramRepository : JpaRepository<Program, Long> {
//    fun findByName(name: String): Program?
}